backuppcl: pass finishesSpec field to resume span

msbutler opened the pull request:
PR #114268 broke the plumbing for the CompletedSpans metric which allows the backup coordinator to update the FractionCompleted metric. This patch fixes this bug by passing the finishesSpec field to the resume span.

Informs #120161

Release note: none

msbutler committed:
backuppcl: pass finishesSpec field to resume span

PR #114268 broke the plumbing for the CompletedSpans metric which allows the
backup coordinator to update the FractionCompleted metric. This patch fixes
this bug by passing the finishesSpec field to the resume span.

Informs #120161

Release note: none

stevendanna commented on pkg/ccl/backupccl/backup_processor.go:
```diff
@@ -636,12 +636,13 @@ func runBackupProcessor(
-								end:        span.end,
-								attempts:   span.attempts,
-								lastTried:  span.lastTried,
+								span:         *resp.ResumeSpan,
+								firstKeyTS:   resumeTS,
+								start:        span.start,
+								end:          span.end,
+								attempts:     span.attempts,
+								lastTried:    span.lastTried,
```
Should we reconsider our pattern her. Perhaps we should copy the whole struct and then reset the fields we want to reset.

msbutler replied:
perhaps in a follow up pr that doesn't get backported?

msbutler replied:
done.

msbutler committed:
backupccl: create resume span by copying old span first

This refactor makes the code less error prone, when adding a new field to
`spanAndTime`.

Release note: none

Epic: none

msbutler commented:
TFTR!

bors r=dt

dt commented:
When you manually backport to 23.1, can you do 23.1.17-rc as well?
